### PR TITLE
Mouse inside contour check

### LINKED ISSUE #15: Check that mouse pos is inside polygons
Since SAM generates masks but Biigle can only deal with polygons this leads to ambiguous results. Add a check that discards polygons that do not contain the mouse position.

### PR DESCRIPTION
**Pull request by dlangenk**
Fixes #15

### CODE REVIEW

File: src/resources/assets/js/ol/MagicSamInteraction.js [hunk 1]
```diff
@@ -143,7 +173,7 @@ class MagicSamInteraction extends PointerInteraction {
             let pointCoordsTensor = new Tensor("float32", pointCoords, [1, 2, 2]);
             const feeds = this._getFeeds(pointCoordsTensor);
 
-            this.model.run(feeds).then(this._processInferenceResult.bind(this));
+            this.model.run(feeds).then(this._processInferenceResult.bind(this,pointCoords));
```
Comment (mzur) [inline src/resources/assets/js/ol/MagicSamInteraction.js:176]:
```suggestion
            this.model.run(feeds).then(this._processInferenceResult.bind(this, pointCoords));
```

File: src/resources/assets/js/ol/MagicSamInteraction.js [hunk 2]
```diff
@@ -209,6 +239,7 @@ class MagicSamInteraction extends PointerInteraction {
 
         let contour = MagicWand.traceContours(imageData)
             .filter(c => !c.inner)
+            .filter(c => isPointInsideContour(c, pointCoords[0], pointCoords[1]))
```
Comment (mzur) [inline src/resources/assets/js/ol/MagicSamInteraction.js:242]: I was going to say that the OpenLayers Polygon geometry has a `intersectsCoordinate()` method that could be used for this. But the coordinates are no Polygon at this point. However, I traced the OpenLayers logic to [this function](https://github.com/openlayers/openlayers/blob/v5.3.0/src/ol/geom/flat/contains.js#L37). Is this the same as yours? Then we could just import it here.

Comment (dlangenk) [reply src/resources/assets/js/ol/MagicSamInteraction.js:242]: If you flatten the contours before then it should be the same.

**Comment by mzur** [comment]
@dlangenk Looks good?

**Comment by dlangenk** [comment]
LGTM

**Pull request merged**